pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -71,18 +88,45 @@ func (q *QueryReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctr
 		return res, err
 	}
 
-	return ctrl.Result{}, q.runListRestorePoints(ctx, req)
+	// Create a TLS secret for the NMA service
+	err := q.generateCertsForNMA(ctx)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	passwd, err := q.GetSuperuserPassword(ctx, q.Log)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	prunner := cmds.MakeClusterPodRunner(q.Log, q.VRec.Cfg, q.Vdb.GetVerticaUser(), passwd)
+	pfacts := MakePodFacts(q.VRec, prunner)
+	dispatcher := q.makeDispatcher(q.Log, q.Vdb, prunner, passwd)
+
+	if e := pfacts.Collect(ctx, q.Vdb); e != nil {
```

Comment:
I got error in the unit tests, "failed to copy and execute the gather script: could not execute: pod vertica-sample-defaultsubcluster-0 does not have a host assigned". It's from this line https://github.com/vertica/vertica-kubernetes/blob/main/pkg/cmds/exec.go#L260. I'm wondering if I've missed any steps for setting up pods after creating the database.